lib/rest-connector.js

```diff
@@ -100,31 +102,12 @@ function RestConnector(baseURL, settings) {
   this._settings = settings || {};
 
   // Extract the default options for request module
-  var options = this._settings.options ||
-    this._settings.defaults || this._settings;
+  var options = omit(defaults(this._settings.options || {},
+                              this._settings,
+                              this._settings.defaults),
+                     ['defaults', 'connector', 'operations']);
```

Comment:
Switched up the precedence order to match this. Current behavior doesn't seem to do any merging, we are just using one object or the other.  In my client cert test I added a check to ensure that the top level settings object wouldn't be completely overwritten by a default or options object.
Take this template for example:
```javascript
var template = {
  cert : 'clientCert.crt',
  defaults : {
     'headers' : { 'my-header' : 'overwrites' }
  }
};
```
You lose the cert attribute because a defaults object is present.  I think merging is reasonable, though the docs should be updated either way if this code isn't accepted to to indicate precedence and if merging happens.